aten/src/ATen/core/IList_inl.h

```diff
@@ -0,0 +1,215 @@
+    const list_type::const_iterator& it
+) {
+  return *it;
+}
+
+inline
+const TORCH_ILIST_IMPL(at::Tensor, Boxed)::list_type&
+TORCH_ILIST_IMPL(at::Tensor, Boxed)::unwrap(
+    const c10::IList<at::Tensor>& ilist
+) {
+  return *ilist.payload_.boxed;
+}
+
+inline
+TORCH_ILIST_IMPL(at::Tensor, Boxed)::list_type::const_iterator&
+TORCH_ILIST_IMPL(at::Tensor, Boxed)::unwrap(
+    c10::IListIterator<at::Tensor>& it
+) {
+  return it.payload_.boxed_iterator;
+}
+
+inline
+const TORCH_ILIST_IMPL(at::Tensor, Boxed)::list_type::const_iterator&
+TORCH_ILIST_IMPL(at::Tensor, Boxed)::unwrap(
+    const c10::IListIterator<at::Tensor>& it
+) {
+  return it.payload_.boxed_iterator;
+}
+
+inline
+c10::detail::IListConstRef<at::Tensor>
+TORCH_ILIST_IMPL(at::Tensor, Boxed)::iterator_get(
+    const list_type::const_iterator& it
+) {
+  return (*it).get().toTensor();
+}
+
+inline
+const TORCH_ILIST_IMPL(at::OptionalTensorRef, Unboxed)::list_type&
+TORCH_ILIST_IMPL(at::OptionalTensorRef, Unboxed)::unwrap(
+    const c10::IList<at::OptionalTensorRef>& ilist
+) {
+  return ilist.payload_.unboxed;
+}
+
+inline
+TORCH_ILIST_IMPL(at::OptionalTensorRef, Unboxed)::list_type::const_iterator&
+TORCH_ILIST_IMPL(at::OptionalTensorRef, Unboxed)::unwrap(
+    c10::IListIterator<at::OptionalTensorRef>& it
+) {
+  return it.payload_.unboxed_iterator;
+}
+
+inline
+const TORCH_ILIST_IMPL(at::OptionalTensorRef, Unboxed)::list_type::const_iterator&
+TORCH_ILIST_IMPL(at::OptionalTensorRef, Unboxed)::unwrap(
+    const c10::IListIterator<at::OptionalTensorRef>& it
+) {
+  return it.payload_.unboxed_iterator;
+}
+
+inline
+c10::detail::IListConstRef<at::OptionalTensorRef>
+TORCH_ILIST_IMPL(at::OptionalTensorRef, Unboxed)::iterator_get(
+    const list_type::const_iterator& it
+) {
+    return *it;
+}
+
+inline
+const TORCH_ILIST_IMPL(at::OptionalTensorRef, Boxed)::list_type&
+TORCH_ILIST_IMPL(at::OptionalTensorRef, Boxed)::unwrap(
+    const c10::IList<at::OptionalTensorRef>& ilist
+) {
+  return *ilist.payload_.boxed;
+}
+
+inline
+TORCH_ILIST_IMPL(at::OptionalTensorRef, Boxed)::list_type::const_iterator&
+TORCH_ILIST_IMPL(at::OptionalTensorRef, Boxed)::unwrap(
+    c10::IListIterator<at::OptionalTensorRef>& it
+) {
+  return it.payload_.boxed_iterator;
+}
+
+inline
+const TORCH_ILIST_IMPL(at::OptionalTensorRef, Boxed)::list_type::const_iterator&
+TORCH_ILIST_IMPL(at::OptionalTensorRef, Boxed)::unwrap(
+    const c10::IListIterator<at::OptionalTensorRef>& it
+) {
+  return it.payload_.boxed_iterator;
+}
+
+inline
+c10::detail::IListConstRef<at::OptionalTensorRef>
+TORCH_ILIST_IMPL(at::OptionalTensorRef, Boxed)::iterator_get(
+    const list_type::const_iterator& it
+) {
+  const auto& ivalue = (*it).get();
+  return (ivalue.isNone()) ? at::OptionalTensorRef{} : ivalue.toTensor();
+}
+
+namespace at {
+// [Note: ITensorList]
+using ITensorList = c10::IList<at::Tensor>;
+using ITensorListIterator = c10::IListIterator<at::Tensor>;
+// [Note: at::OptionalTensorRefRefList]
```

Comment:
Nit: this doesn't match `[Note: IOptTensorRefList]` in the codegen below

Reply:
Ah, you are right. Nice catch!